src/test/java/com/github/dockerjava/netty/exec/ExecStartCmdExecTest.java

```diff
@@ -193,16 +197,22 @@ public void execStartNotAttachedStdin() throws Exception {
         ExecCreateCmdResponse execCreateCmdResponse = dockerClient.execCreateCmd(container.getId())
                 .withAttachStdout(true)
                 .withAttachStdin(false)
-                .withCmd("/bin/sh").exec();
+                .withCmd("/bin/sh")
+                .exec();
 
         boolean completed = dockerClient.execStartCmd(execCreateCmdResponse.getId())
                 .withDetach(false)
                 .withStdIn(stdin)
                 .exec(new ExecStartResultCallback(stdout, System.err))
                 .awaitCompletion(5, TimeUnit.SECONDS);
 
-        // with v1.22 of the remote api the server closed the connection when no stdin was attached while exec create, so completed was true
-        assertFalse(completed, "The process was not finished.");
         assertEquals(stdout.toString(), "");
+
+        if (getVersion(dockerClient).isGreaterOrEqual(VERSION_1_23)) {
```

Comment:
Actually it was all around test code base. During refactoring i touch this but without clear understanding. But if the same behavior changes withing API versions it can be used but the best way to create parametrized test with API version as param (tags if fine too, like with cicleci)